dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/FlowReduxStateMachine.kt

```diff
@@ -62,14 +57,12 @@ public abstract class FlowReduxStateMachine<S : Any, A : Any>(
 
     override suspend fun dispatch(action: A) {
         checkSpecBlockSet()
-        activeFlowCounterMutex.withLock {
```

Comment:
I'm not 100% sure what the purpose of the mutex _here_ exactly is. As far as I can see, it has 2 relevant effects:

1. It makes so that if the exception gets thrown, that happens-before the next increment or decrement.
2. It makes so that reading the value happens-after the previous inc/dec - i.e. the value update gets actually seen in the next block. I.e. the effect you'd get on pure jvm by marking the var as volatile.

It seems to me that only 2 is relevant here, and that gets taken care of by using the AtomicInt.